Design options tests DNM for feedback

mjkkirschner opened the pull request:
@Pboyer,
apologies for the strange diff setup here, I wanted to save you the time looking at code you've already seen, so this diff only includes: 
11 tests 
4 dyn files
and changes I've made to log to the console while loading the serialized presets

does not include any of the Spike work for the states editor

mjkkirschner commented on test/DynamoCoreTests/PresetStatesTests.cs:
```diff
@@ -0,0 +1,301 @@
+        }
+
+          [Test]
+          [Category("Failure")]
```
this TODO and failure category is erroneous, this has been fixed, will remove

riteshchandawar commented:
@holyjewsus It will be good if you add any TD to the PR which is related to Test Case addition so that we will be knowing which kind of test cases we are having in our system.

Thanks,
Ritesh

mjkkirschner commented:
@riteshchandawar sorry about that! Will do from now on.

pboyer commented:
@holyjewsus This is a good start.  The one thing to point out is you should use `DynamoModelTestBase` wherever possible as your base class.  It looks like you don't really need the `ViewModel` layer for any of this, so why include it?  

Note, also, that we will be removing all WPF dependencies from `DynamoCoreTests`, so this will be necessary to bring in your code.

mnhng committed:
OpenFileDialog implementation

aparajit-pratap committed:
fix to overwrite excel files having a single sheet

monikaprabhu committed:
Address issues in PR review for Excel tests automation

aparajit-pratap committed:
minor code refactor

monikaprabhu committed:
add test for formula in Excel

monikaprabhu committed:
Fix file path

zora-wang committed:
Merge branch 'master' of https://github.com/DynamoDS/Dynamo

zora-wang committed:
add issue ID for failure tests

aparajit-pratap committed:
fixed element rewriter for global functions and global classes

Commit:
Merge remote-tracking branch 'remotes/upstream/master' into InCanvasSearch-ReturnToSeachTextBox

Commit:
Merge remote-tracking branch 'remotes/upstream/master' into SearchVM_take_only_DynamoVM

Commit:
Revert AssemblyFileVersion.

Commit:
Revert AssemblyVersion 2 time.

aparajit-pratap committed:
cleaned up excel test cases to reuse existing test APIs

aparajit-pratap committed:
Merge pull request #4617 from aparajit-pratap/cleanExcelTests

Cleaned up excel test cases to reuse existing test APIs

Commit:
Tooltip added to InCanvasSearch.

ramramps committed:
Merge pull request #4580 from aosyatnik/SearchVM_take_only_DynamoVM

MAGN-7450: SearchViewModel constructor should only take DynamoViewModel

mjkkirschner committed:
before modifying any node when setting the state of the graph, we can set the state x,y of the node to the current x,y so the node is not moved

add a test for this behavior

Commit:
InCanvas tooltip is aligned at the right and left places.

mjkkirschner committed:
add you track link

kronz committed:
Update README.md

ramramps committed:
Merge pull request #4583 from ramramps/Graph-Annotation-Title-Zoom-Edit-fix

MAGN-7070 Can't see group title editing when zoomed out

mjkkirschner committed:
Merge pull request #4584 from holyjewsus/LibraryToolTipPositionMAGN7397

Library tool tip position incorrect on high dpi screens, magn7397

monikaprabhu committed:
Add missing files
Fix tests based on PR review comments

zora-wang committed:
update

zora-wang committed:
Merge branch 'master' of https://github.com/DynamoDS/Dynamo

chuanyuee committed:
Merge pull request #4594 from Randy-Ma/master

Part 1 of fixing MAGN-7251 [Regression from 0.7.5] Select elements will cause a crash after switching Revit documents

monikaprabhu committed:
WriteToReadOnlyFile() this test is not consistently working correct

monikaprabhu committed:
remove unnecessary line of code
Replace one more test with assertPreview value

monikaprabhu committed:
Merge branch 'master' of https://github.com/DynamoDS/Dynamo into test

aparajit-pratap committed:
Merge pull request #4512 from monikaprabhu/test

Automation for Excel tests

mnhng committed:
update libG for CoordinateSystem changes

aparajit-pratap committed:
added check to test case

Commit:
Do not use canvas context menu by Name, instead use sender.

aparajit-pratap committed:
addressed review comments

aparajit-pratap committed:
minor code refactor

ramramps committed:
Merge pull request #4608 from aosyatnik/InCanvasSearch-ReturnToSeachTextBox

InCanvasSearch: return focus to seach text box

Commit:
Test for InCanvasTooltip added.

Commit:
Merge remote-tracking branch 'remotes/upstream/master' into InCanvasSearch-Tooltip

Commit:
Exit added.

Commit:
Remove Run and Exit.

mjkkirschner committed:
Merge branch 'upmaster' into designOptionsTests

Conflicts:
	src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs

performed a merge, only conflict was autogenned info file,

lukechurch committed:
TT Templates can't have spaces at the end of their line

lukechurch committed:
Merge pull request #4626 from DynamoDS/lukechurch_ttTemplateFix

TT Templates can't have spaces at the end of their line

zora-wang committed:
Merge branch 'master' of https://github.com/DynamoDS/Dynamo

aparajit-pratap committed:
Merge pull request #4616 from aparajit-pratap/elementRewrite

Fixed element rewriter for global functions and global classes

ke-yu committed:
Enable node to code

ke-yu committed:
Merge pull request #4628 from ke-yu/enable-n2c

Enable node to code

aparajit-pratap committed:
corrected minor mistakes in gallery content

aparajit-pratap committed:
minor changes

aparajit-pratap committed:
minor changes

zora-wang committed:
Merge branch 'master' of https://github.com/DynamoDS/Dynamo

sharadkjaiswal committed:
Merge pull request #4611 from nguyen-binh-minh/dialog

MAGN-7016: OpenFileDialog

pbidenko committed:
Merge branch 'master' of https://github.com/DynamoDS/Dynamo

aparajit-pratap committed:
minor refactor

aparajit-pratap committed:
Merge pull request #4615 from aparajit-pratap/fixExcelOverwrite

Fix to overwrite excel files having a single sheet

zora-wang committed:
Merge branch 'master' of https://github.com/DynamoDS/Dynamo

Commit:
Remove ScrollViewer from IncanvasSearchControl.

pbidenko committed:
REACH-268 Reach: All node preview values are initially shown as "null"

kronz committed:
Merge pull request #4629 from aparajit-pratap/galleryContent

Corrected minor mistakes in gallery content

mjkkirschner committed:
when packageContents property notification is raised we need to raise a can execute changed event on the submitCommand, this is raised when package contents changed when custom nodes or dlls are added

chuanyuee committed:
Merge pull request #4623 from nguyen-binh-minh/syncLibG

update libG for CoordinateSystem changes

ke-yu committed:
Move strings to resource file

ke-yu committed:
Update strings

ke-yu committed:
Make resource public and use in python script editor

ke-yu committed:
Merge pull request #4639 from ke-yu/script-editor-localization

Globalize Python Script Editor

ke-yu committed:
Remove Validity.Assert()

ke-yu committed:
Add test cases

mnhng committed:
localize operation fail messages

mnhng committed:
localize Excel errors

aparajit-pratap committed:
updated broken test case

aparajit-pratap committed:
Merge pull request #4642 from aparajit-pratap/updateExcelTest

Updated broken Excel test

mnhng committed:
localize some Engine warnings

ke-yu committed:
Merge pull request #4643 from nguyen-binh-minh/localization

MAGN-7532: Localization of Excel and Engine warnings

pbidenko committed:
EventHandler -> Action<HomeWorkspaceModel>

Commit:
Delete test.

Commit:
OnMembersListBoxMouseWheel added in IncanvasSearchControl. Now VerticalOffset is calculated manually.

Commit:
New converter is added to change the height of context menu in workspace view.

pboyer committed:
Merge pull request #4634 from pbidenko/REACH-268

Reach 268: All node preview values are initially shown as "null"

Commit:
Use BringIntoView for new highlighted item.

mjkkirschner committed:
added a test, it uses reflection, not sure if this is acceptable.

mjkkirschner committed:
Merge branch 'upmaster' into designOptionsTests

Conflicts:
	src/DynamoCoreWpf/Properties/Resources.en-US.resx
	src/DynamoCoreWpf/Properties/Resources.resx

merge and regain resources designer

zora-wang committed:
update recorded test

Sancerio committed:
fix MAGN-7473

mnhng committed:
move resource strings from ProtoCore to DSOffice

sharadkjaiswal committed:
Merge pull request #4651 from Sancerio/master

fix MAGN-7473

mnhng committed:
make resources public

Commit:
Misprint is corrected.

Commit:
Fix input port of CBN for localized build

- Unbound variable name was obtained from warning message and for
localized builds the warning messages are localized, so it is difficult
to parse it correctly to get the variable name.
- Use UnboundVariableSymbolNode property of WarningEntry object to get
the unbound variable name.

zora-wang committed:
upload CSV file

Commit:
Categories alphabetical order removed from SearchViewModel.

monikaprabhu committed:
Merge pull request #4502 from zora-wang/master

add recording test for colin workshop files

sharadkjaiswal committed:
Merge pull request #4656 from Conceptual-Design/master

Fix input port of CBN for localized build

aparajit-pratap committed:
rewrite default argument expressions with fully qualified names

aparajit-pratap committed:
removed unused code

aparajit-pratap committed:
changed access specifier

mnhng committed:
MAGN-7533

benglin committed:
Updated a few incorrect conditional statements

mnhng committed:
migrate Rectangle.ByWidthHeight to Rectangle.ByWidthLength

sharadkjaiswal committed:
Merge pull request #4663 from nguyen-binh-minh/MAGN-7398

MAGN7398 - migrate Rectangle.ByWidthHeight to Rectangle.ByWidthLength

benglin committed:
Merge pull request #4662 from Benglin/magn7573

Fixed crashes in Reorder and RemoveKey nodes

Commit:
Added check for null.

Commit:
ListBox changed into TreeView.

monikaprabhu committed:
Merge branch 'master' of https://github.com/DynamoDS/Dynamo into test

sharadkjaiswal committed:
Merge pull request #4654 from nguyen-binh-minh/master

move resource strings from ProtoCore to DSOffice

zora-wang committed:
add testcase

mnhng committed:
resync libG

aparajit-pratap committed:
added tests

aparajit-pratap committed:
removed unused references, using statements

benglin committed:
Merge pull request #4668 from nguyen-binh-minh/gg

resync libG

Commit:
Removed key navigation.

mnhng committed:
temporary fix for CBN test case

sharadkjaiswal committed:
Merge pull request #4669 from nguyen-binh-minh/gg

temporary fix for CBN test case

Commit:
separate Package Manager from DynamoCore

zora-wang committed:
Merge branch 'master' of https://github.com/DynamoDS/Dynamo into coordinateSystemDefaultArgument

zora-wang committed:
add description in test

zora-wang committed:
add test case to verify connections are not swapping

pboyer committed:
Merge pull request #4646 from aosyatnik/InCanvasSearch-ScrollingFix

InCanvasSearch: scrolling fix

monikaprabhu committed:
Merge branch 'master' of https://github.com/DynamoDS/Dynamo into test

benglin committed:
Fixed wonky execution issues

Commit:
Merge remote-tracking branch 'remotes/upstream/master' into LibrarySearch-XAMLChanges

sharadkjaiswal committed:
Merge pull request #4674 from Benglin/MAGN7575

Fixed wonky execution issues

Commit:
minor fixes & tests for deserialization check

Commit:
More button removed.

Commit:
missed including of projects in the solution

ramramps committed:
Merge pull request #4666 from aosyatnik/LibrarySearch-XAMLChanges

Library Search: XAML changes.

mjkkirschner committed:
get rid of reflection call to addFile

don't use begin invoke to call raise execute change when adding custom nodes to package content, instead just raise it

change addfile to internal

pboyer committed:
Merge pull request #4676 from aosyatnik/RemoveMoreButton

Remove "More" button

mjkkirschner committed:
Merge branch 'upmaster' into designOptionsTests

merged latest

kronz committed:
added description of new IF node behavior

aparajit-pratap committed:
addressed review comments

Sancerio committed:
Resize truncated buttons width

benglin committed:
Merge pull request #4679 from Sancerio/master

Resize truncated buttons width

mnhng committed:
MAGN-7541

benglin committed:
Merge pull request #4681 from nguyen-binh-minh/localization

localization issues

mnhng committed:
MAGN-7643

sharadkjaiswal committed:
Merge pull request #4683 from nguyen-binh-minh/master

MAGN-7643 Point.BySphericalCoordinate tooltip

mnhng committed:
MAGN-7649

mnhng committed:
localize Select node control

aparajit-pratap committed:
addressed review comments

aparajit-pratap committed:
removed commented code

mnhng committed:
localize Watch node, String node and File node controls

mjkkirschner committed:
removed test from UI tests, added to new class to avoid synchronization issue

pboyer committed:
Merge pull request #4665 from aosyatnik/LibrarySearch-CategoryAlphabeticalOrder

Library Search: remove alphabetical order in categories

mjkkirschner committed:
change name and remove un-used usings

mjkkirschner committed:
Merge branch 'upmaster' into PublishButtonFix

merge latest

mjkkirschner committed:
this test now fails on the CI, trying to figure out why.

Commit:
Fix conflicts with LibG updates and version skew.

Merge branch 'master' into MAGN-6964

Conflicts:
	extern/LibG_219/LibG.AsmPreloader.Managed.dll
	extern/LibG_219/LibG.AsmPreloader.Unmanaged.dll
	extern/LibG_219/LibG.Managed.dll
	extern/LibG_219/LibG.ProtoInterface.dll
	extern/LibG_219/LibG.dll
	extern/LibG_219/LibGCore.dll
	extern/LibG_220/LibG.AsmPreloader.Managed.dll
	extern/LibG_220/LibG.AsmPreloader.Unmanaged.dll
	extern/LibG_220/LibG.Managed.dll
	extern/LibG_220/LibG.ProtoInterface.dll
	extern/LibG_220/LibG.dll
	extern/LibG_220/LibGCore.dll
	extern/LibG_221/LibG.AsmPreloader.Managed.dll
	extern/LibG_221/LibG.AsmPreloader.Unmanaged.dll
	extern/LibG_221/LibG.Managed.dll
	extern/LibG_221/LibG.ProtoInterface.dll
	extern/LibG_221/LibG.dll
	extern/LibG_221/LibGCore.dll
	extern/ProtoGeometry/ProtoGeometry.XML
	extern/ProtoGeometry/ProtoGeometry.dll
	extern/ProtoGeometry/ProtoInterface.dll
	extern/ProtoGeometry/en-US/ProtoGeometry.resources.dll
	src/DynamoCore/Core/PreferenceSettings.cs
	src/DynamoCoreWpf/Properties/Resources.Designer.cs
	src/DynamoCoreWpf/Properties/Resources.en-US.resx
	src/DynamoCoreWpf/Properties/Resources.resx
	src/Libraries/Analysis/PointData.cs
	src/Libraries/Analysis/SurfaceData.cs
	src/Libraries/Analysis/VectorData.cs

Commit:
Merge pull request #4686 from ikeough/MAGN-6964

Magn 6964

Commit:
Merge pull request #4621 from aosyatnik/InCanvasSearch-Tooltip

InCanvasSearch: tooltip

Commit:
Import updated Helix with selection variables in shaders.

Commit:
Remove PerVertexMeshGeometryModel3D

We will use DynamoGeometryModel3D from our fork of Helix instead.

Commit:
Use two meshes.

mjkkirschner committed:
Merge pull request #4635 from holyjewsus/PublishButtonFix

PublishButton greyed out - MAGN 7569

monikaprabhu committed:
Automation for MAGN-7573

monikaprabhu committed:
Merge branch 'master' of https://github.com/DynamoDS/Dynamo into test

sharadkjaiswal committed:
Merge pull request #4691 from nguyen-binh-minh/localization

Localization issues

Commit:
InCanvasSearch width is bind to context menu width.

aparajit-pratap committed:
Merge pull request #4659 from aparajit-pratap/defaultArgResolve

Rewrite default argument expressions with fully qualified names

benglin committed:
Merge pull request #4647 from aosyatnik/InCanvasSearch-Jump

InCanvasSearch: jump

monikaprabhu committed:
Merge pull request #4671 from zora-wang/coordinateSystemDefaultArgument

test coordinatesystem

monikaprabhu committed:
Merge pull request #4672 from zora-wang/assignValueToVariable

add test case to verify connectors are not swapping

zora-wang committed:
add imagePixel testcase

Commit:
Detach Model3D objects and null collections.

Commit:
Remove memory leak caused by circular reference.

In DynamoView, I had created a binding to a property on an
VisualizationSettingsViewModel instance which held a reference back to
the view model.

Commit:
Convert all tests to test render packages.

Commit:
Cleanup Watch3DView.

Commit:
Detach and unsubscribe on shutdown.

Commit:
Remove performance info from sln.

mjkkirschner committed:
made most changes from Peter's review, need to look into Zero Touch hardening comment

mjkkirschner committed:
Merge branch 'upmaster' into designOptionsTests

Conflicts:
	src/DynamoCoreWpf/Properties/Resources.Designer.cs
	src/DynamoCoreWpf/Properties/Resources.en-US.resx
	src/DynamoCoreWpf/Properties/Resources.resx

fixed merge conflicts in resx files

mjkkirschner committed:
fixed bad merge between resource file, en was missing a key, so did not notice during merge. fixed.

mjkkirschner committed:
instead of modifying zero touch base function which explicitly checks that document path is not null, I introduced a new save context for presets which avoid the save context.file path in zero touch nodes.  I think this is more clear and keeps the zero touch serialization failing where it should.

Sancerio committed:
fix UI truncation on Publish Package button

benglin committed:
Merge pull request #4700 from Sancerio/master

fix UI truncation on Publish Package button

zora-wang committed:
localization for ZoomToFit

ke-yu committed:
Merge pull request #4688 from monikaprabhu/test

Automation for bug MAGN-7573

zora-wang committed:
give description to node ExportToSAT

sharadkjaiswal committed:
Merge pull request #4704 from zora-wang/local4

give description to node ExportToSAT

Commit:
Default category for custom nodes has been added.

monikaprabhu committed:
Merge pull request #4695 from zora-wang/imagePixelBranch

add imagePixel testcase

Commit:
Merge pull request #4687 from ikeough/MAGN-7462

MAGN-7462 Selection of nodes does not consistently highlight geometry

Commit:
Experimental search algorithm created.

Commit:
Merge branch 'master' into RenderingMemoryImprovements

Conflicts:
	src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs
	test/DynamoCoreWpfTests/VisualizationManagerUITests.cs

Commit:
Remove showShadows and binding.

Commit:
Fix show edges binding.

Commit:
Rename this file to be consistent with class.

Commit:
Merge pull request #4698 from ikeough/RenderingMemoryImprovements

Rendering Memory Improvements

ramramps committed:
Fix for DummyNodes GUID's

Fix for DummyNodes GUID's

pboyer committed:
First steps for package manager fix

pboyer committed:
Properly provider root package directory to PackageManagerClient

Commit:
Remove dlls.

Commit:
Put dlls back.

Commit:
Merge pull request #4711 from ikeough/HelixFix

Helix fix

Commit:
Stop rendering if render caps are not adequate.

Commit:
Merge pull request #4713 from ikeough/RenderCapsCheck

Stop rendering if render caps are not adequate.

ke-yu committed:
Merge pull request #4702 from zora-wang/master

localization for ZoomToFit

mnhng committed:
change the locale of libG on startup

monikaprabhu committed:
Merge branch 'master' of https://github.com/DynamoDS/Dynamo into test

monikaprabhu committed:
fix guid in AssertPreview

monikaprabhu committed:
Fix the GUid in AssertPreview Value.

ke-yu committed:
Merge pull request #4718 from monikaprabhu/test

Fix test   Removekey_7573()

benglin committed:
Merge pull request #4714 from nguyen-binh-minh/libG

change the locale of libG on startup

aparajit-pratap committed:
updated LibG binaries

aparajit-pratap committed:
Merge pull request #4721 from aparajit-pratap/libGbin

Updated LibG binaries from code fixes in LibG

ramramps committed:
Added a regression test

Added a regression test

ramramps committed:
Merge pull request #4709 from ramramps/MAGN-7568

MAGN-7568 Cannot open dyn file from German user

Commit:
minor fixes & tests

aparajit-pratap committed:
updated LibG binaries

aparajit-pratap committed:
Merge pull request #4724 from aparajit-pratap/master

Updated LibG binaries

pboyer committed:
Merge pull request #4705 from aosyatnik/CategoryForCustomNodes

Default category for custom nodes

kronz committed:
Merge pull request #4710 from pboyer/pkgfix

Fixes for publish package failure.  We need to make sure this gets cherry-picked to 0.8.1 after testing in master.

ke-yu committed:
Insert parameter for instance method and property

ke-yu committed:
Add test case

riteshchandawar committed:
Added test case for defect MAGN-7635

riteshchandawar committed:
updated test case correctly.

ke-yu committed:
Merge pull request #4728 from riteshchandawar/TestForDefect_7635

Added test case for defect MAGN-7635

Sancerio committed:
Fix unable to find localized GalleryContents issue

benglin committed:
Merge pull request #4730 from Sancerio/master

Fix unable to find localized GalleryContents issue

benglin committed:
Disabled RunSettingsDisableRun test case

benglin committed:
Merge pull request #4732 from Benglin/master

Disabled RunSettingsDisableRun test case

Commit:
minor fixes chapter 2

Commit:
Merge branch 'master' of https://github.com/DynamoDS/Dynamo into package-manager

Conflicts:
	src/DynamoCore/Models/DynamoModel.cs

Commit:
Merge branch 'master' of https://github.com/DynamoDS/Dynamo into package-manager

Conflicts:
	src/DynamoCore/Models/DynamoModel.cs

Commit:
Merge branch 'package-manager' of https://github.com/marimano/Dynamo into package-manager

mjkkirschner committed:
remove presetModel class - rename PresetState -> PresetModel

workspaceModel just holds a list of PresetModel and exposes Ienum<PresetModel>

NodeGraph now stores Presets and deserializes them, deserialization logic has been moved from PresetModel collection to each Preset through a static constructor that node graph calls

no longer create new presetsModels on new workspaces, instead just supply empty Enum<PresetModel>

no longer implement logger for presetsCollection since it does not exist

a tiny bit more logic on graph serialization to serialize each preset into a "Presets" element , workspace now also gets add and remove state methods

added a redo test

mjkkirschner committed:
fix test dyns to use new element names

mjkkirschner committed:
renaming

refactor LINQ statement

white space cleanup

mjkkirschner committed:
name changes, and changing some bindings that strangely were not updated...

mjkkirschner committed:
make a few changes considering custom nodes more clearly

ke-yu committed:
Update test case Callsite_RunWithTraceDataFromUnresolvedNodes_DoesNotCrash

ke-yu committed:
Remove comments

sharadkjaiswal committed:
Merge pull request #4736 from ke-yu/fix_testcase

Fix testcase Callsite_RunWithTraceDataFromUnresolvedNodes_DoesNotCrash

ke-yu committed:
Minor change to avoid duplication

ke-yu committed:
Merge pull request #4729 from ke-yu/magn7574

Fix magn-7574: Custom node creation with curve input sets type to "double"

mnhng committed:
migrate Rectangle.ByWidthHeight to Rectangle.ByWidthLength

sharadkjaiswal committed:
Merge pull request #4741 from nguyen-binh-minh/master

migrate Rectangle.ByWidthHeight to Rectangle.ByWidthLength

mnhng committed:
fix parameters description not used in tooltip and help dialog

ke-yu committed:
Merge pull request #4744 from nguyen-binh-minh/MAGN-7642

fix parameters description not used in tooltip and help dialog

Commit:
include new-added projects into solutions

ke-yu committed:
Mark test case as failure

sharadkjaiswal committed:
Merge pull request #4746 from ke-yu/fix_testcase

Mark test case AddingDyfRaisesCanExecuteChangeOnDelegateCommand as failure

pboyer committed:
Merge pull request #4707 from aosyatnik/PluralSearch

New search algorithm (experimental, don't merge)

mjkkirschner committed:
custom node type parameters don't have a function descriptor so they can't be used to lookup an assembly or for description,summary etc...

mjkkirschner committed:
Merge pull request #4750 from holyjewsus/TypeParameterbuildfix

merging this in, it clears up some failing tests locally

pboyer committed:
Merge pull request #4670 from marimano/package-manager

MAGN-7280 Move package manager code out of DynamoCore, make an Extension

mjkkirschner committed:
Merge branch 'upmaster' into designOptionsTests

no merge conflicts?

mjkkirschner committed:
renaming changes

mjkkirschner committed:
change order of presets in constructors for homeworkspace model and custom nodes

update names from states -> presets in UI

mjkkirschner committed:
modify AddPreset to take enum of guids
change addpreset command so that it derives from modelBase rec command to reuse logic for serializing lots of guids - this was failing sometimes otherwise

fix missing cases in recordable command deserialization

add preset does not need a list of guids, or an enum
